client/src/store/login/actions.ts

```diff
@@ -1,54 +1,10 @@
-import API from '../api';
-import {store} from '../../App';
-import {
-  LOGIN_CHANGE_EMAIL_TEXT,
-  LOGIN_CHANGE_PASSWORD_TEXT,
-  LOGIN_PENDING,
-  LOGIN_SUCCESS,
-  LOGIN_ERROR
-} from './actionTypes';
+import API from "../api";
+const TOKEN = "TOKEN";
 
-const TOKEN = 'TOKEN';
-
-export const setEmailText = (email: string) => ({
-  type: LOGIN_CHANGE_EMAIL_TEXT,
-  payload: email
-});
-
-export const setPasswordText = (password: string) => ({
-  type: LOGIN_CHANGE_PASSWORD_TEXT,
-  payload: password
-});
-
-export const loginPending = () => ({
-  type: LOGIN_PENDING
-});
-
-export const loginSuccess = () => ({
-  type: LOGIN_SUCCESS
-});
-
-export const loginError = (error: any) => ({
-  type: LOGIN_ERROR,
-  payload: error
-});
-
-export const getApiData = () => {
-  return (dispatch: Function) => {
-    const {login: {email, password}} = store.getState();
-    dispatch(loginPending());
-    API.post(('/auth/login'), {
-      email,
-      password
+export const loginUser = (user: object) => (dispatch: Function) => {
+  return API.post("/auth/login", user)
```

Comment:
where is actions? how would you follow the flow?
Where is PENDING, SUCCESS and ERROR?
why do you need redux?

Reply:
Instead of PENDING and SUCCESS actions I'm using redux-form's built-in 'submitting' props. It tells whether or not form is currently submitting. It will be true until the promise is resolved or rejected. And untill it's true, I'm showing a loading spinner.
When there is an error, i will use showAlert action, which will show it on the screen.